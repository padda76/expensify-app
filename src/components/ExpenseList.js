import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem {...expense} key={expense.id} />;
                })
            )
        }

    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses,state.filters)
    };
}

export default connect(mapStateToProps)(ExpenseList);

//default connected version to be used i app
//named export to be used in tests