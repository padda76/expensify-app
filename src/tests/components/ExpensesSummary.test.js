import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from './../fixtures/expenses';

test('should render summary of 1 expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render summary of multiple expenses correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={45} expensesTotal={98756623} />);
    expect(wrapper).toMatchSnapshot();
});