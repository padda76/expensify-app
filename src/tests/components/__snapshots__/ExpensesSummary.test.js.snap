// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary of 1 expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h1>
</div>
`;

exports[`should render summary of multiple expenses correctly 1`] = `
<div>
  <h1>
    Viewing 
    45
     
    expenses
     totalling 
    $987,566.23
  </h1>
</div>
`;
